internal/services/amlfilesystem/aml_file_system_resource.go

```diff
@@ -0,0 +1,512 @@
+}
+
+type AMLFileSystemResource struct{}
+
+var _ sdk.ResourceWithUpdate = AMLFileSystemResource{}
+
+func (r AMLFileSystemResource) ResourceType() string {
+	return "azurerm_aml_file_system"
+}
+
+func (r AMLFileSystemResource) ModelObject() interface{} {
+	return &AMLFileSystemModel{}
+}
+
+func (r AMLFileSystemResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return amlfilesystems.ValidateAmlFilesystemID
+}
+
+func (r AMLFileSystemResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.AMLFileSystemName,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"maintenance_window": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"day_of_week": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringInSlice(amlfilesystems.PossibleValuesForMaintenanceDayOfWeekType(), false),
+					},
+
+					"time_of_day_in_utc": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validate.TimeOfDayInUTC,
+					},
+				},
+			},
+		},
+
+		"sku_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"storage_capacity_in_tb": {
+			Type:     pluginsdk.TypeInt,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.IntBetween(8, 128),
+				validation.IntDivisibleBy(8),
+			),
+		},
+
+		"subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"zones": commonschema.ZonesMultipleRequiredForceNew(),
+
+		"hsm_setting": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			ForceNew: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"container_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: storageValidate.StorageContainerResourceManagerID,
+					},
+
+					"logging_container_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: storageValidate.StorageContainerResourceManagerID,
+					},
+
+					"import_prefix": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ForceNew:     true,
+						ValidateFunc: validate.ImportPrefix,
+					},
+				},
+			},
+		},
+
+		"identity": commonschema.UserAssignedIdentityOptionalForceNew(),
+
+		"key_encryption_key": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
```

Comment:
Can we check whether users can delete the key after they added it? If not, then shall we mark this as conditional force new ([example](https://github.com/hashicorp/terraform-provider-azurerm/blob/4868c47a6205ccf0d92a08a7fad0c480521bda84/internal/services/compute/snapshot_resource.go#L115-L122))? Otherwise, we shall consider switching the patch update to be the full update, as the patch update can not afford to support removing the key?

Reply:
Thanks. I marked it as conditional force new since the update operation and the createOrUpdate operation can't remove keyEncryptionKey.